packages/storage-plus/src/stack.rs

```diff
@@ -0,0 +1,649 @@
+    /// StdError::NotFound if there is no item with that key
+    ///
+    /// # Arguments
+    ///
+    /// * `storage` - a reference to the storage this item is in
+    /// * `key` - a byte slice representing the key to access the stored item
+    fn load_impl(&self, storage: &dyn Storage, key: &[u8]) -> StdResult<T> {
+        let prefixed_key = [self.as_slice(), key].concat();
+        must_deserialize(&Some(
+            storage
+                .get(&prefixed_key)
+                .ok_or(StdError::not_found(type_name::<T>()))
+                .unwrap(),
+        ))
+    }
+
+    /// Returns StdResult<()> resulting from saving an item to storage
+    ///
+    /// # Arguments
+    ///
+    /// * `storage` - a mutable reference to the storage this item should go to
+    /// * `key` - a byte slice representing the key to access the stored item
+    /// * `value` - a reference to the item to store
+    fn save_impl(&self, storage: &mut dyn Storage, key: &[u8], value: &T) -> StdResult<()> {
+        let prefixed_key = [self.as_slice(), key].concat();
+        storage.set(&prefixed_key, &to_vec(value)?);
+        Ok(())
+    }
+
+    /// Returns StdResult<()> resulting from saving an item to storage
+    ///
+    /// # Arguments
+    ///
+    /// * `storage` - a mutable reference to the storage this item should go to
+    /// * `key` - a byte slice representing the key to access the stored item
+    /// * `value` - a reference to the item to store
+    fn remove_impl(&self, storage: &mut dyn Storage, key: &[u8]) -> StdResult<()> {
+        let prefixed_key = [self.as_slice(), key].concat();
+        storage.remove(&prefixed_key);
+        Ok(())
+    }
+}
+
+/// An iterator over the contents of the Stack store.
+pub struct StackIter<'a, T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    stack: &'a Stack<'a, T>,
+    storage: &'a dyn Storage,
+    start: u32,
+    end: u32,
+}
+
+impl<'a, T> StackIter<'a, T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    /// constructor
+    pub fn new(stack: &'a Stack<'a, T>, storage: &'a dyn Storage, start: u32, end: u32) -> Self {
+        Self {
+            stack,
+            storage,
+            start,
+            end,
+        }
+    }
+}
+
+impl<'a, T> Iterator for StackIter<'a, T>
+where
+    T: Serialize + DeserializeOwned,
+{
+    type Item = StdResult<T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.start >= self.end {
+            return None;
+        }
+        let item = self.stack.get_at(self.storage, self.start).unwrap();
+        self.start += 1;
+        Some(Ok(item))
+    }
+
+    // This needs to be implemented correctly for `ExactSizeIterator` to work.
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        let len = (self.end - self.start) as usize;
+        (len, Some(len))
+    }
+
+    // I implement `nth` manually because it is used in the standard library whenever
```

Comment:
Please rewrite those comments - refrain from writing in first-person.

```suggestion
    // `nth` is implemented manually, because it is used in the standard library whenever
```